include/ode_solvers/collocation.h

```diff
@@ -0,0 +1,357 @@
+/*
+GeomScale Project
+
+Copyright (c) 2020
+  Vissarion Fisikopoulos
+  Apostolos Chalkis
+  Elias Tsigaridas
+  Marios Papachristou
+
+Contributed and/or modified by Marios Papachristou,
+as part of Google Summer of Code 2020 program.
+
+VolEsti is free software: you can redistribute it and/or modify it
+under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or (at
+your option) any later version.
+
+VolEsti is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+for more details.
+
+See the file COPYING.LESSER for the text of the GNU Lesser General
+Public License.  If you did not receive this file along with HeaDDaCHe,
+see <http://www.gnu.org/licenses/>.
+
+Resource: https://en.wikipedia.org/wiki/Collocation_method
+
+*/
+
+#ifndef COLLOCATION_H
+#define COLLOCATION_H
+
+#include <csignal>
+
+
+template <typename Point, typename NT, class Polytope, class bfunc, class func=std::function <Point(std::vector<Point>, NT)>>
+class CollocationODESolver {
+public:
+
+  // Vectors of points
+  typedef std::vector<Point> pts;
+  typedef std::vector<pts> ptsv;
+
+  // typedef from existing templates
+  typedef typename Polytope::MT MT;
+  typedef typename Polytope::VT VT;
+  typedef std::vector<MT> MTs;
+  typedef std::vector<func> funcs;
+  typedef std::vector<Polytope*> bounds;
+  typedef std::vector<NT> coeffs;
+
+  unsigned int dim;
+
+  NT eta;
+  NT t, t_prev, dt;
+  const NT tol = 1e-6;
+
+  // If set to true the solver assumes linearity of the field
+  // Otherwise it approximates the constant vector with Euler method
+  const bool exact = false;
+
+  // Boundary oracle method
+  std::string boundary_oracle_method;
+
+  // If set to true it enables precomputation (does not recompute A and b at every step)
+  const bool precompute = true;
+
+  bool precompute_flag = false;
+
+  // Function oracles x'(t) = F(x, t)
+  funcs Fs;
+
+  // Basis functions
+  bfunc phi, grad_phi;
+
+  bounds Ks;
```

Comment:
Most of them follow the mathematical notation. We can change them to be more descriptive.
I personally prefer using the math symbols as they are when writing code that is correlated to maths directly,
since one can look at the papers and find out about the code directly. From a sweng perspective, I acknowledge
that it is not the best practice.

Reply:
ok, I think I prefer the mathematical notation, so if you agree please add some comment such as the variable names use the math notation from paper [..]

Reply:
Ok I added a general description on the main header file `include/ode_solvers/ode_solvers.hpp`.